### PR TITLE
[ALLUXIO-1879] Improve AlluxioURI with additional functionality

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-1879

### CODE REVIEW

File: core/common/src/main/java/alluxio/AlluxioURI.java [hunk 1]
```diff
@@ -11,20 +11,21 @@
...
-import java.net.URISyntaxException;
 import java.util.Arrays;
+import java.util.Map;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * It uses a hierarchical URI internally. URI requires that String is escaped, {@link AlluxioURI}
- * does not.
+ * It uses a {@link URI} internally. URI requires that String is escaped, {@link AlluxioURI} does
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/AlluxioURI.java]: Please add a sentence explaining the purpose for this class, before jumping into how it works internally.

**Comment by jsimsa** [comment]
@gpang this is great, thank you very much for adding support for queries a generalizing AlluxioURI to multi-schemed URIs

File: core/common/src/main/java/alluxio/AlluxioURI.java (hunk 1 above)
Comment (gpang) [inline core/common/src/main/java/alluxio/AlluxioURI.java]: Updated the comment.

**Comment by gpang** [comment]
@jsimsa Thanks for the comment! I updated the PR.

**Comment by jsimsa** [comment]
LGTM

File: core/common/src/main/java/alluxio/URI.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+ *     * D:\path\to\file
+ *     * E:\path\to\skip\..\file
+ *   * URI with multiple scheme components
+ *     * scheme://host:123/path
+ *     * scheme:part2://host:123/path
+ *     * scheme:part2:part3://host:123/path
+ *
+ * Currently, does not support fragment in the URI.
+ */
+public interface URI extends Comparable<URI>, Serializable {
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/URI.java]: Is it confusing if we overload `URI`?

Comment (gpang) [inline core/common/src/main/java/alluxio/URI.java]: @calvinjia That is a good question. Well, currently, nothing other than our AlluxioURI uses this interface, and I don't foresee anything other than AlluxioURI to use it. What do you think it should be renamed to?

Comment (calvinjia) [inline core/common/src/main/java/alluxio/URI.java]: Makes sense, Java URI is an actual implementation so it should be clear enough.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**